src/parser/monk/mistweaver/modules/spells/SoothingMist.js

```diff
@@ -33,55 +31,61 @@ class SoothingMist extends Analyzer {
   constructor(...args) {
     super(...args);
     this.assumedGCD = 1500 / (1 + this.statTracker.hastePercentage(this.statTracker.currentHasteRating)) *.95;
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.SOOTHING_MIST), this.castSoothingMist);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.VIVIFY), this.castSoothingMistVivify);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.ENVELOPING_MIST), this.castSoothingMistEnvelopingMist);
```

Comment:
You could combine both of these into 1 action listener and 1 function since the functions are the exact same
something like
`this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell([SPELLS.ENVELOPING_MIST, SPELLS.VIVIFY]), this.castDuringSoothingMist);

Reply:
Thanks. I didn't realize we could track 2 or more spells per listener. Updated in newest push